### ISSUE TITLE
Questions about training Supernet in PostNAS

### ISSUE THREAD

**Issue by XuTongxue12138** [NVlabs/Jet-Nemotron#18]
Hello,I learned a lot from reading your paper. I have a few questions about the details below. 
a). I would like to ask about the training details of Supernet during the first stage of PostNAS, when exploring the importance of the full attention layer. In this stage, is the sampling path the same as in the subsequent beam search, where only one full attention layer is selected, and the remaining layers use linear attention layers, or are all selected randomly?
b). In your paper, during the beam search phase, you mentioned replacing each layer with a full attention layer and the remaining layers with linear attention layers. For each specific task, is the number of architectures in the search space limited to the number of layers in the pre-trained model?
c). What is the ultimate goal you want to achieve when training Supernet? Is it to make the output of any sampled path as similar as possible to the output of the previously pre-trained model?
d). How many times did you train the system throughout the experiment, and in which parts? Did you train the PostNAS component four times, and then restart the two-stage training process after obtaining the final architecture?

**Comment by t1101675** [comment]
(a) Layers are selected randomly during training.
(b) The search space consists of all potential combinations of linear and full attention layers. Because searching this entire space is infeasible, we use beam search.
(c) Yes.
(d) Yes.

**Comment by XuTongxue12138** [comment]
> (a) Layers are selected randomly during training. (b) The search space consists of all potential combinations of linear and full attention layers. Because searching this entire space is infeasible, we use beam search. (c) Yes. (d) Yes.

Okay, thank you.I have a few more questions to ask
e) During training, your super network is composed of sub networks formed by randomly sampling paths at each step. How many random sampling paths have you performed in total
f) How did you determine that the super network containing $2 ^ {28}$subnetworks really achieved the desired performance? Did you randomly select paths to activate the subnetworks and verify that they met the requirements n times? If so, may I ask what are the confidence levels for n and satisfying the requirements
g) Does the 'once for all' super network you mentioned here mean that finding the architecture only requires training once, and when increasing or decreasing the number of full attention layers in the future, simply perform beam search in this super network instead of deploying the idea directly after one training session

**Comment by t1101675** [comment]
(e) We did not control the exact sampling paths. We just randomly sample a path in each step.
(f) We measure the validation loss of the supernetwork by randomly sampling the paths.
(g) Yes

**Comment by XuTongxue12138** [comment]
> (e) 我们没有控制具体的采样路径，只是在每一步中随机选择一条路径进行采样。 (f) 我们通过随机采样路径来测量超网络的验证损失。 (g) 是的

Okay, thank you for your reply. May I ask what is the ratio of the training set to the validation set when training a supernet

**Comment by OSCAR-WU2019** [comment]
> (e) We did not control the exact sampling paths. We just randomly sample a path in each step. (f) We measure the validation loss of the supernetwork by randomly sampling the paths. (g) Yes

Hi authors, thanks for your excellent work ! I have a few questions to clarify details about the training process

I am also confused about question (f) and would like to confirm two sub-points:
(f.1) Does the statement "We measure the validation loss of the supernetwork by randomly sampling the paths" mean you randomly sample paths and calculate their mean validation loss to evaluate how well the model is trained?
(f.2) Are there more details about the first-stage training (e.g., how many epochs did you train for, and what criteria did you use to stop training)?
Could you please provide some explanations? Thank you very much!

**Comment by t1101675** [comment]
f.1 Yes
f.2 We train the model with 50B tokens. We stop the training when we observe that the MMLU performance does not grow with training.

**Comment by OSCAR-WU2019** [comment]
OK,thanks. I have few more questions to clarify the key details about the supernet construction and its training process, which will help us better understand and reproduce your work.

(1) Through our research, we found that some studies adopt the weight sharing strategy to construct a supernet (e.g., the standard attention module and the linear attention module use the same parameters, with their parameters kept consistent and updated synchronously during training). We wonder if this weight sharing strategy is applied in your work?
 Or are the parameters of the standard attention module and linear attention module in the supernet   two completely independent sets of parameters (They are updated independently of each other, and may be initialized with the same parameters)?

(2) If the weight sharing strategy is adopted, do you use an additional pre-trained model as the teacher model to guide the training of the supernet (i.e., guiding the parameter updates of both the standard attention module and linear attention module in the supernet)?

(3) For the active path randomly sampled during training, are only the parameters of the linear attention module updated (with the parameters of the standard attention module frozen throughout the training), or are the parameters of both the linear attention and standard attention modules within this path updated?

(4) According to the statement in the paper: "During training,we randomly sample an active path at each step, forming a subnetwork, which is trained using feature distillation loss [46, 47, 48]". Was only the feature distillation loss used to train the sampled subnetwork in the first-stage training? Or was the training conducted with a combined loss of cross-entropy loss and distillation loss, following a design paradigm similar to Once for all paper?
